components/script/dom/webglrenderingcontext.rs

```diff
@@ -1281,15 +1290,26 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
             return self.webgl_error(InvalidOperation);
         }
 
-        if count <= 0 {
-            return self.webgl_error(InvalidOperation);
+        if count < 0 {
+            return self.webgl_error(InvalidValue);
         }
 
         if offset < 0 {
             return self.webgl_error(InvalidValue);
         }
 
-        if self.current_program.get().is_none() || self.bound_buffer_element_array.get().is_none() {
+        if self.current_program.get().is_none() {
```

Comment:
Also, for the record, I'm not suggesting "don't check the program", but just return early without generating an error, since at least when there's no bound program that's what the spec says.